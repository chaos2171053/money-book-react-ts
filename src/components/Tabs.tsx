import React, { Component } from 'react'

interface TabProps {
    children: React.ReactNode
}
interface TabsProps {
    activeIndex: number,
    children: React.ReactNode,
    onTabChange: Function
}

interface Istate {
    activeIndex: number
}


export const Tab = (props: TabProps) => (
    <React.Fragment>{props.children}</React.Fragment>
)

export class Tabs extends Component<TabsProps, Istate> {
    constructor(props: TabsProps) {
        super(props);
        this.state = {
            activeIndex: props.activeIndex
        }
    }
    tabChange = ((event: MouseEvent, index: number) => {
        event.preventDefault()
        this.setState({
            activeIndex: index
        })
        this.props.onTabChange(index)
    })
    render() {
        const { children } = this.props
        const { activeIndex } = this.state
        return (
            <ul className="tabs-component nav nav-tabs nav-fill my-4">
                {React.Children.map(children, (child, index) => {
                    const activeClassName = (activeIndex === index) ? 'nav-link active' : 'nav-link'
                    return (
                        <li className="nav-item">
                            <a
                                onClick={(event: any) => { this.tabChange(event, index) }}
                                className={activeClassName}
                                role="button"
                                href='javascript;'
                            >
                                {child}
                            </a>
                        </li>
                    )
                })}
            </ul>
        )
    }
}