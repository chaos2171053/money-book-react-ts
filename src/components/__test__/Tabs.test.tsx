import React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import { Tabs, Tab } from '../Tabs'


let props = {
    activeIndex: 0,
    onTabChange: jest.fn()
}
let wrapper: ReactWrapper
describe('test Tabs compotent', () => {
    beforeEach(() => {
        wrapper = mount(
            <Tabs {...props}>
                <Tab>这是第一个Tab</Tab>
                <Tab>这是第二个Tab</Tab>
            </Tabs>)
    })
    it('should render the component to match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should render correct tab item', () => {
        expect(wrapper.find('Tab').length).toEqual(2)
    })
})