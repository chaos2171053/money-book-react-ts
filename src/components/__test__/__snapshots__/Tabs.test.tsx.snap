// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs compotent should render the component to match snapshot 1`] = `ReactWrapper {}`;
